tests/terraform/graph/variable_rendering/test_foreach_renderer.py

```diff
@@ -573,3 +575,21 @@ def test_double_nested_foreach_and_count_with_variable_reference():
                                        'module.level1["green"].module.level2["test1.txt"].aws_s3_bucket_object.this_file[1]',
                                        'module.level1["blue"].module.level2["test2.txt"].aws_s3_bucket_object.this_file[1]',
                                        'module.level1["green"].module.level2["test2.txt"].aws_s3_bucket_object.this_file[1]']
+
+
+@mock.patch.object(env_vars_config, "RAW_TF_IN_GRAPH_ENV", "True")
+def test_foreach_renderer_with_raw_asset():
```

Comment:
maybe also check the configuration changes accordingly? (i.e `resource.config[virtual_resources] = ...`)